bpf/bpf_sock.c

```diff
@@ -335,6 +339,12 @@ static __always_inline int __sock4_xlate_fwd(struct bpf_sock_addr *ctx,
 	if (sock4_skip_xlate(svc, orig_key.address))
 		return -EPERM;
 
+#ifdef ENABLE_LOCAL_REDIRECT_POLICY
+	if (lb4_svc_is_localredirect(svc) &&
```

Comment:
Do we have an idea how costly this extra lookup is? We could consider making this code-path conditional on whether or not skip map entries exist?

Reply:
>whether or not skip map entries exist

Wouldn't that extra call be comparable to doing a look up in the map?

Anyway I don't think it's necessary, the lookup is only done for LRP services, and that too at the socket (`connect`) calls.

Reply:
Oh right, yeah if it's part of `connect` then there's no point to optimize this away.